packages/protocol/contracts/governance/DowntimeSlasherIntervals.sol

```diff
@@ -0,0 +1,409 @@
+pragma solidity ^0.5.3;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./SlasherUtil.sol";
+
+contract DowntimeSlasherIntervals is SlasherUtil {
+  using SafeMath for uint256;
+
+  // Intervals previously slashed
+  struct SlashedInterval {
```

Comment:
I will keep it to check overlapping just in case the SlashableDowntime changes in the future. Otherwise we could have problems if we calculated every time, the endBlock.
And we should check that range because we could have an epoch crossing interval, that will be assigned to the StartBlock epoch, but if the next epoch is still down, it could be slashed again

Reply:
Just saw the other comments.